table/metadata.go

```diff
@@ -215,6 +218,7 @@ func MetadataBuilderFromBase(metadata Metadata) (*MetadataBuilder, error) {
 func (b *MetadataBuilder) HasChanges() bool { return len(b.updates) > 0 }
 
 func (b *MetadataBuilder) CurrentSpec() iceberg.PartitionSpec {
+	// FIXME: this should return a pointer which can be nil to avoid out of bounds indexing / np derefs
 	return b.specs[b.defaultSpecID]
```

Comment:
can we fix this now instead of leaving the `FIXME`?

Reply:
Taking another look at this fn - b.specs is not a map and specs may get added and removed as we go, so it's very much possible to be in a state where SpecID = 1023 but b.specs has only length of 2

Reply:
uses b.GetSpecByID(b.defaultSpecID) instead now